src/main/java/org/opensearch/security/configuration/SettingsPermissionValveImpl.java

```diff
@@ -0,0 +1,206 @@
+    private final ClusterService clusterService;
+    private final AdminDNs adminDNs;
+    private final AuditLog auditLog;
+
+    public SettingsPermissionValveImpl(
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        AdminDNs adminDNs,
+        AuditLog auditLog
+    ) {
+        this.clusterService = clusterService;
+        this.adminDNs = adminDNs;
+        this.auditLog = auditLog;
+
+        // Add listener for configuration updates
+        clusterService.addListener(event -> {
+            SecurityDynamicConfiguration<RoleV7> config = rolesConfiguration.get();
+            if (config != null) {
+                // Handle any cluster state related updates if needed
+                // TODO: is this needed here? It's used in DlsFlsValveImpl but seems to be working well without it here on cluster updates
+            }
+        });
+    }
+
+    @Override
+    public boolean invoke(PrivilegesEvaluationContext context, ActionListener<?> listener) {
+        final ActionRequest request = context.getRequest();
+        
+        // Skip validation for admin users
+        if (adminDNs.isAdmin(context.getUser())) {
+            return true;
+        }
+
+        try {
+            if (request instanceof ClusterUpdateSettingsRequest) {
+                return validateClusterSettings(context, (ClusterUpdateSettingsRequest) request, listener);
+            } else if (request instanceof UpdateSettingsRequest updateSettingsRequest) {
+                return validateIndexSettings(context, updateSettingsRequest.settings(), updateSettingsRequest, listener);
+            } else if (request instanceof CreateIndexRequest createIndexRequest) {
+                return validateIndexSettings(context, createIndexRequest.settings(), createIndexRequest, listener);
+            }
+            return true;
+        } catch (Exception e) {
+            log.error("Error while evaluating settings permissions", e);
+            listener.onFailure(new OpenSearchSecurityException("Error while evaluating settings permissions: " + e.getMessage(), RestStatus.FORBIDDEN));
+            return false;
+        }
+    }
+
+    private boolean validateClusterSettings(
+        PrivilegesEvaluationContext context,
+        ClusterUpdateSettingsRequest request,
+        ActionListener<?> listener
+    ) {
+        // Get allowed settings patterns from user's roles
+        Set<String> allowedSettings = getAllowedSettingsFromRoles(context);
+
+        // For backwards compatibility we will allow all settings if no allowed settings are defined
+        if (allowedSettings.isEmpty()) {
+            return true;
+        }
+        log.debug("Allowed settings: {} for user: {}", allowedSettings, context.getUser().getName());
+
+        // Validate persistent settings
+        if (!validateSettingsMap(request.persistentSettings(), allowedSettings)) {
+            auditLog.logMissingPrivileges(context.getAction(), request, context.getTask());
```

Comment:
I guess we need to introduce a new audit logging category here. This way, it would look in the audit logs like the user is missing the action privilege indicated by `context.getAction()`.